core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -49,6 +53,47 @@ public GraphDatabaseAPI getDb() {
         return db;
     }
 
+    private boolean awaitUntil(BooleanSupplier supplier, int maxTime) {
+        int time = 0;
+        int sleep = 100;
+        boolean condition = supplier.getAsBoolean();
+
+        while (!condition && time < maxTime) {
+            Util.sleep(sleep);
+            time += sleep;
+            condition = supplier.getAsBoolean();
+        }
+
+        return condition;
+    }
+
+
+    private void waitForSystemDbLifecycles() {
+        /* Await 2 min at max for the system database to show in
+           status CURRENT or REQUIRES_UPGRADE.
+           We do not do anything for the other possible status because
+           the dbms would not even be able to operate normally in them
+           (i.e. UNSUPPORTED_BUT_CAN_UPGRADE, UNSUPPORTED, UNSUPPORTED_FUTURE).
+
+           The reason for that is that the first time we initialize
+           it the built-in roles (and possibly other system databases
+           structures) will not be populated immediately in 4.4.
+           I.e. system.isAvailable can output true but some of the
+           Lifecycles may not have been run yet (UNINITIALIZED status)
+         */
+        boolean condition = awaitUntil( () ->
+                                        {
+                                            var components = dependencyResolver.resolveDependency( SystemGraphComponents.class );
+                                            var status = components.detect( db );
+                                            return (status == Status.CURRENT || status == Status.REQUIRES_UPGRADE);
+                                        }, 120000 );
```

Comment:
I guess the fail-save solution for this could be to parametrize this value, so users can extend the time frame if needed. What do you think?

Reply:
2 minutes should be more than enough for an instance to have run all of the lifecycles. I feel like introducing a new parameter for this is an overkill